torch/utils/data/dataloader.py

```diff
@@ -90,9 +91,9 @@ def default_collate(batch):
             storage = batch[0].storage()._new_shared(numel)
             out = batch[0].new(storage)
         return torch.stack(batch, 0, out=out)
-    elif type(batch[0]).__module__ == 'numpy':
+    elif elem_type.__module__ == 'numpy' and elem_type.__name__ != 'str_':
```

Comment:
what about general object arrays?

Reply:
Good point

Reply:
General object arrays (and numpy arrays that contain objects) won't go into this case because elem_type.__module__ won't be 'numpy', it'll be related to the object. They'll fall through to the end of this function which will raise a TypeError

Reply:
Huh? `numpy` arrays that contain objects have different `__module__` than `numpy`? O_o that's unexpected

Reply:
Oh okay, I see what you mean. Sorry I was confused because the code checks the type of the first element of batch (`batch[0]`), not the type of batch itself. Fix incoming.